src/controllers/features/unincorporated/addressCorrespondanceSelectorController.ts

```diff
@@ -26,7 +26,7 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
         // set addressoption to render the page with saved data
         let addressOption = "";
         const applicantDetails = acspData.applicantDetails || {};
-        if (applicantDetails.correspondenceAddress !== null) {
+        if (applicantDetails.correspondenceAddress !== null && applicantDetails.correspondenceAddress !== undefined) {
```

Comment:
(applicantDetails.correspondenceAddress != null) will check for both null and undefined
or (applicantDetails.correspondenceAddress) will check all falsey values

Reply:
Its not checking thats why the bug was there.